packages/eslint-plugin/src/rules/no-misused-spread.ts

```diff
@@ -0,0 +1,185 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import * as tsutils from 'ts-api-utils';
+import * as ts from 'typescript';
+
+import {
+  createRule,
+  getConstrainedTypeAtLocation,
+  getParserServices,
+  isBuiltinSymbolLike,
+  isTypeFlagSet,
+} from '../util';
+
+type MessageIds =
+  | 'noStringSpreadInArray'
+  | 'noSpreadInObject'
+  | 'noFunctionSpreadInObject';
+
+export default createRule<[], MessageIds>({
+  name: 'no-misused-spread',
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'Disallow using the spread operator when it might cause unexpected behavior',
+      recommended: 'strict',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      noStringSpreadInArray:
+        "Using the spread operator on a string is not allowed in an array. Use `String.split('')` instead.",
+
+      noSpreadInObject:
+        'Using the spread operator on `{{type}}` is not allowed in an object.',
+
+      noFunctionSpreadInObject:
+        'Using the spread operator on `Function` without properties is not allowed in an object. Did you forget to call the function?',
```

Comment:
> without properties

Without ... _added_ properties? Some might point out that functions do have built-in properties.

Reply:
haha sure, good idea